packages/sync-actions/src/order-actions.js

```diff
@@ -37,3 +38,56 @@ export function actionsMapDeliveries(diff, oldObj, newObj) {
 
   return handler(deliveriesDiff, oldObj.shippingInfo, newObj.shippingInfo)
 }
+
+export function actionsMapReturnsInfo(diff, oldObj, newObj) {
+  const returnInfoDiff = diff.returnInfo
+  if (!returnInfoDiff) return []
+
+  const handler = createBuildArrayActions('returnInfo', {
+    [CHANGE_ACTIONS]: (oldSReturnInfo, newReturnInfo) => {
+      const updateActions = Object.keys(diff.returnInfo).reduce(
+        (itemActions, key) => {
+          const { items = {} } = diff.returnInfo[key]
+          if (Object.keys(items).length > 0) {
+            return [
+              ...itemActions,
+              ...Object.keys(items).reduce((actions, index) => {
+                if (items[index].shipmentState) {
```

Comment:
Im not sure if this is a legit case, however, this will fail if we want to generate both an `setReturnShipmentState` and `setReturnPaymentState`. For example,

```js
const before = {
  returnInfo: [
    {
      returnTrackingId: "touched-item",
      items: [
        {
          id: "test-1",
          shipmentState: "returned",
          paymentState: "initial"
        }
      ]
    }
  ]
}

const now = {
  returnInfo: [
    {
      returnTrackingId: "touched-item",
      items: [
        {
          id: "test-1",
          shipmentState: "backInStock",
          paymentState: "refunded"
        }
      ]
    }
  ]
}

// will always only return the `setReturnShipmentState`
[
  {
    action: "setReturnShipmentState",
    returnItemId: "test-1",
    shipmentState: "backInStock"
  }
]

```

Reply:
Nice catch. It is possible to modify both. Thanks for pointing that out 💪